internal/controller/rotators/aws_oidc_rotator.go

```diff
@@ -0,0 +1,199 @@
+	ctx context.Context
+	// client is used for Kubernetes API operations.
+	client client.Client
+	// kube provides additional Kubernetes API capabilities.
+	kube kubernetes.Interface
+	// logger is used for structured logging.
+	logger logr.Logger
+	// stsOps provides AWS STS operations interface.
+	stsOps STSClient
+	// backendSecurityPolicyName provides name of backend security policy.
+	backendSecurityPolicyName string
+	// backendSecurityPolicyNamespace provides namespace of backend security policy.
+	backendSecurityPolicyNamespace string
+	// preRotationWindow specifies how long before expiry to rotate.
+	preRotationWindow time.Duration
+}
+
+// NewAWSOIDCRotator creates a new AWS OIDC rotator with the specified configuration.
+// It initializes the AWS STS client and sets up the rotation channels.
+func NewAWSOIDCRotator(
+	ctx context.Context,
+	client client.Client,
+	kube kubernetes.Interface,
+	logger logr.Logger,
+	backendSecurityPolicyNamespace string,
+	backendSecurityPolicyName string,
+	preRotationWindow time.Duration,
+	region string,
+) (*AWSOIDCRotator, error) {
+	cfg, err := defaultAWSConfig(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load AWS config: %w", err)
+	}
+
+	cfg.Region = region
+
+	if proxyURL := os.Getenv("AI_GATEWAY_STS_PROXY_URL"); proxyURL != "" {
+		cfg.HTTPClient = &http.Client{
+			Transport: &http.Transport{
+				Proxy: func(*http.Request) (*url.URL, error) {
+					return url.Parse(proxyURL)
+				},
+			},
+		}
+	}
+
+	stsClient := NewSTSClient(cfg)
+
+	return &AWSOIDCRotator{
+		client:                         client,
+		kube:                           kube,
+		logger:                         logger.WithName("aws-oidc-rotator"),
+		stsOps:                         stsClient,
+		backendSecurityPolicyNamespace: backendSecurityPolicyNamespace,
+		backendSecurityPolicyName:      backendSecurityPolicyName,
+		preRotationWindow:              preRotationWindow,
+	}, nil
+}
+
+// SetSTSOperations sets the STS operations implementation - primarily used for testing.
+func (r *AWSOIDCRotator) SetSTSOperations(ops STSClient) {
+	r.stsOps = ops
+}
+
+// IsExpired checks if the preRotation time is before the current time.
+func (r *AWSOIDCRotator) IsExpired() bool {
+	preRotationExpirationTime := r.GetPreRotationTime()
+	return IsBufferedTimeExpired(0, preRotationExpirationTime)
+}
+
+// UpdateCtx is used to update the context used in AWSOIDCRotator functions.
+// This can be used to set timeouts for outgoing calls to assume role.
+func (r *AWSOIDCRotator) UpdateCtx(ctx context.Context) {
+	r.ctx = ctx
+}
+
+// GetPreRotationTime gets the expiration time minus the preRotation interval or return zero value for time.
+func (r *AWSOIDCRotator) GetPreRotationTime() time.Time {
+	secret, err := LookupSecret(r.ctx, r.client, r.backendSecurityPolicyNamespace, GetBSPSecretName(r.backendSecurityPolicyName))
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return time.Time{}
+		}
+		return time.Time{}
+	}
```

Comment:
wouldn't you want to distinguish the error getting secret vs IsExpired returning true at the callsite? I would suggest you return error from this function simply
```suggestion
func (r *AWSOIDCRotator) GetPreRotationTime() time.Time {
	secret, err := LookupSecret(r.ctx, r.client, r.backendSecurityPolicyNamespace, GetBSPSecretName(r.backendSecurityPolicyName))
	if err != nil {
		if !errors.IsNotFound(err) {
			return time.Time{}
		}
		return time.Time{}
	}
```